### PR TITLE
basic service setup

### PR DESCRIPTION
**Pull request by KevinLe563**
PR for settings up services

### CODE REVIEW

**Comment by KevinLe563** [comment]
for adding/modifying future services should we branch off this child branch or just merge this one and then branch off main?

File: include/service/telemetry_collection_service.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/**
```
Comment (dgobalak) [inline include/service/telemetry_collection_service.h:1]: Could you add an include guard for this too?

File: include/service/system_status_service.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
```
Comment (dgobalak) [inline include/service/system_status_service.h:1]: Could you add an include guard for this header file?

File: include/service/system_status_service.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+typedef struct {
+	// TODO: Add important system health parameters
+} sys_health_t;
+
+/**
+  Reboots the system.
+  Should only be used if necessary.
+  @return 1 if success, 0 on fail
+*/
+int sys_reboot(void);
```
Comment (dgobalak) [inline include/service/system_status_service.h:18]: I'd use uint8_t as the return type for all of these functions.

File: include/service/system_status_service.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+  Changes state of satellite to emergency mode in the case of an error.
+  Should be used purely as a state changer
+  @return 1 if successful state change, else 0 on fail
+*/
+int sys_emergency_state();
```
Comment (dgobalak) [inline include/service/system_status_service.h:25]: Is there a reason why this function doesn't have a void argument, but the one above does?

**Review by dgobalak — CHANGES_REQUESTED** [review]

File: include/service/system_status_service.h (hunk 3 above)
Comment (KevinLe563) [reply include/service/system_status_service.h:18]: Just followed the csp practice of just using int, not sure if uint8_t is preferred?

Comment (dgobalak) [reply include/service/system_status_service.h:21]: Interesting. Ya if csp uses int for their return types, I guess it's better to stay consistent (unless we wanted to rewrite csp lol).

File: include/service/telemetry_collection_service.h [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+*/
+
+/**
+  Converts the struct to a character buffer
+  @param struct TODO: needs to be a specific struct
+  @param inbuf user provided buffer for storing converted struct
+  @param inlen length of expected buffer after conversion
+  @return 1 or buffer size on success, 0 on failure
+*/
+int struct_to_char_buff(void * inbuf, int inlen);
```
Comment (dgobalak) [inline include/service/telemetry_collection_service.h:51]: I'd make the inlen parameter type uint8_t. I think csp uses uint for their parameters when needed.

Comment (dgobalak) [reply include/service/telemetry_collection_service.h:51]: Idk what the max value for inlen is tho, so maybe uint32_t or something would be better.

Comment (dgobalak) [reply include/service/telemetry_collection_service.h:51]: And this applies to the outlen parameter in the char_buff_to_struct function too.

File: include/service/telemetry_collection_service.h [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+// TODO: Send and collect telemetry should be similar to
+// existing csp functions. Should use existing functions
+// when possible
+
+// TODO: Figure out if this is necessary or if built in
+// get command in sfsf will suffice
+/**
+  Sends a request to submodule, collects telemetry and store in a buffer.
+  @param TODO: Add parameters (port for connection, buffer etc.)
+  @return 0 if successful, else error code
```
Comment (dgobalak) [inline include/service/telemetry_collection_service.h:28]: In the future, we could make a return typedef for error codes.

File: include/service/system_status_service.h (hunk 3 above)
Comment (KevinLe563) [reply include/service/system_status_service.h:21]: We'll keep it as int for now, we can just change it later if necessary

File: include/service/telemetry_collection_service.h (hunk 5 above)
Comment (KevinLe563) [reply include/service/telemetry_collection_service.h]: Yeah thats fair, csp uses int for their outbuf and inbuf lengths so I don't want to mix the types but i guess we could change csp to use uint8/32

File: include/service/telemetry_collection_service.h (hunk 6 above)
Comment (KevinLe563) [reply include/service/telemetry_collection_service.h]: just going to add a TODO for now

File: include/service/telemetry_collection_service.h [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+* Ensures data transmitted at the right time.
+*/
+
+// CSP Includes
+#include <csp/csp.h>
+
+/**
+ * Maps each subsystem to a port
+ */
+enum subsystem_port_t {
```
Comment (dgobalak) [inline include/service/telemetry_collection_service.h:15]: Is there a reason why there isn't a typedef for this?

**Review by dgobalak — APPROVED** [review]
Looks good to me.

**Review by CyStorm — APPROVED** [review]

**Pull request merged**